### ISSUE TITLE
[FR] - dedicated SEO field

### ISSUE THREAD

**Issue by piotrpog** [vaersaagod/seomate#29]
I think seomate would benefit from the dedicated SEO field. I know that it might be against plugin's philosophy, but hear me out. 

While I used seomate in my recent projects, I always set up the same set of fields, which is a bit of the chore (well I know I can use project config file for that but this is not always possible). And native Craft fields don't always have the same functionality as dedicated SEO fields. For example, SEO title field may show you how many characters are left from 70 characters allowed for title, but still allow you to enter more.

Along SEO field, there could also be some general SEO setings section in the control panel, with basic stuff like global description, global SEO image etc. This might be disabled by default to avoid breaking things upon update because people get their global settings from other sources already.

To sum it up - I like how seomate uses config files to decide which craft fields are source of SEO values. But plugin would be even better if it offered some dedicated SEO fields in the control panel.

**Comment by mmikkel** [comment]
Hi @piotrpog, thanks for contributing (and sorry for the late reply).

We're not going to add a custom fieldtype to SEOMate. This plugin was designed to be nothing but a minimal - but super flexible - layer on top of your content, and adding a custom field type to the mix would run counter to that goal, while (in our opinion) adding very little benefit.

In our opinion, most - if not all - of whatever functionality a dedicated SEO field can offer, either a) can easily be created using either core fields or existing third party fieldtypes, or b) would not be worth the trade-off in terms of having your project's dependency on SEOMate increase _substantially_ due to your content now living in an "SEOMate" custom fieldtype.

I'm locking this issue, but leaving it open, since this is a fairly common feature request and it's good to keep this answer around as a reference.

PS:

To speak to your specific example about having a character count/soft limit – that could easily be solved by using a Redactor field for the SEO title, configured to use the [Counter](https://imperavi.com/redactor/plugins/counter/) Redactor plugin (which is included in your Redactor Craft plugin installed) and a short field instruction informing the user that the title should ideally be below 70 characters.

And, as for general/global SEO settings - typically, we create fields for the default/global SEO data, and add those fields to a dedicated "SEO" Global Set. Then, we configure the `defaultMeta` to use those fields.
